File: src/js/bingNews/models/trendingTopics.ts
import {createNewsTopicFromDiscriminatorValue} from './createNewsTopicFromDiscriminatorValue';
import {Answer, NewsTopic} from './index';
import {Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class TrendingTopics extends Answer implements Parsable {
    /** A list of trending news topics on Bing */
    private _value?: NewsTopic[] | undefined;
    /**
     * Instantiates a new TrendingTopics and sets the default values.
     */
    public constructor() {
        super();
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {...super.getFieldDeserializers(),
            "value": n => { this.value = n.getCollectionOfObjectValues<NewsTopic>(createNewsTopicFromDiscriminatorValue); },
        };
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        super.serialize(writer);
        writer.writeCollectionOfObjectValues<NewsTopic>("value", this.value);
    };
    /**
     * Gets the value property value. A list of trending news topics on Bing
     * @returns a NewsTopic
     */
    public get value() {
        return this._value;
    };
    /**
     * Sets the value property value. A list of trending news topics on Bing
     * @param value Value to set for the value property.
     */
    public set value(value: NewsTopic[] | undefined) {
        this._value = value;
    };
}

File: src/dotnet/bingNews/Bing/Models/TrendingTopics.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
namespace Bing.Models {
    public class TrendingTopics : Answer, IParsable {
        /// <summary>A list of trending news topics on Bing</summary>
        public List<NewsTopic> Value { get; set; }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        /// </summary>
        public static new TrendingTopics CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new TrendingTopics();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public new IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>>(base.GetFieldDeserializers()) {
                {"value", n => { Value = n.GetCollectionOfObjectValues<NewsTopic>(NewsTopic.CreateFromDiscriminatorValue)?.ToList(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        /// </summary>
        public new void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            base.Serialize(writer);
            writer.WriteCollectionOfObjectValues<NewsTopic>("value", Value);
        }
    }
}
